File: verification/verify_final.py
import time
import subprocess
import re
from playwright.sync_api import sync_playwright, expect

def verify():
    # Start server
    server = subprocess.Popen(['python3', '-m', 'http.server', '8082'])
    time.sleep(2)
<<<<<<< Updated upstream

=======

>>>>>>> Stashed changes
    try:
        with sync_playwright() as p:
            browser = p.chromium.launch(headless=True)
            context = browser.new_context()
            page = context.new_page()
<<<<<<< Updated upstream

            # Bypass intro
            page.add_init_script("localStorage.setItem('syndicate_live', 'true')")

            # 1. Verify Sticky Widget on Optimization page
            print("Navigating to optimization page...")
            page.goto('http://localhost:8082/optimization.html')

            widget = page.locator('#readiness-command')

            # Scroll down
            page.evaluate("window.scrollTo(0, 1000)")
            time.sleep(0.5)

            # Check compact state (regex)
            expect(widget).to_have_class(re.compile(r"compact"))
            print("Sticky widget verified.")

            # 2. Verify Matrix Gutter Sync on Index page
            print("Navigating to index page...")
            page.goto('http://localhost:8082/index.html')

            # Take screenshots in both modes
            page.screenshot(path='verification/final_dark_gutters.png')
            print("Dark mode screenshot captured.")

=======

            # Bypass intro
            page.add_init_script("localStorage.setItem('syndicate_live', 'true')")

            # 1. Verify Sticky Widget on Optimization page
            print("Navigating to optimization page...")
            page.goto('http://localhost:8082/optimization.html')

            widget = page.locator('#readiness-command')

            # Scroll down
            page.evaluate("window.scrollTo(0, 1000)")
            time.sleep(0.5)

            # Check compact state (regex)
            expect(widget).to_have_class(re.compile(r"compact"))
            print("Sticky widget verified.")

            # 2. Verify Matrix Gutter Sync on Index page
            print("Navigating to index page...")
            page.goto('http://localhost:8082/index.html')

            # Take screenshots in both modes
            page.screenshot(path='verification/final_dark_gutters.png')
            print("Dark mode screenshot captured.")

>>>>>>> Stashed changes
            # Toggle light mode
            page.click('#theme-toggle')
            time.sleep(0.5)
            page.screenshot(path='verification/final_light_gutters.png')
            print("Light mode screenshot captured.")
<<<<<<< Updated upstream

=======

>>>>>>> Stashed changes
            print("Verification successful.")
            browser.close()
    finally:
        server.kill()

if __name__ == "__main__":
    verify()

File: verification/verify_final.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Bypass intro
  await page.addInitScript(() => {
    localStorage.setItem('syndicate_live', 'true');
  });

  // Start server
  const { spawn } = require('child_process');
  const server = spawn('python3', ['-m', 'http.server', '8081']);

  await new Promise(resolve => setTimeout(resolve, 2000));

  try {
    console.log("Navigating to optimization page...");
    await page.goto('http://localhost:8081/optimization.html');

    // Check Sticky
    const widget = await page.locator('#readiness-command');
    const initialBox = await widget.boundingBox();
    console.log(`Initial Widget Y: ${initialBox.y}`);

    await page.evaluate(() => window.scrollTo(0, 1000));
    await page.waitForTimeout(500);

    const isCompact = await widget.evaluate(el => el.classList.contains('compact'));
    const position = await widget.evaluate(el => getComputedStyle(el).position);
    console.log(`Scrolled: isCompact=${isCompact}, position=${position}`);

    // Check Matrix Background Sync
    await page.goto('http://localhost:8081/index.html');
    const matrixBg = await page.evaluate(() => {
        // We can't easily check canvas state, but we can check the matrixBgColor variable if exposed,
        // or just ensure no errors and the canvas exists.
        return typeof matrixBgColor !== 'undefined' ? matrixBgColor : 'not_exposed';
    });
    console.log(`Matrix Background Variable: ${matrixBg}`);

    await page.screenshot({ path: 'verification/final_dark.png' });

    await page.click('#theme-toggle');
    await page.waitForTimeout(200);
    await page.screenshot({ path: 'verification/final_light.png' });

  } catch (e) {
    console.error(e);
  } finally {
    server.kill();
    await browser.close();
  }
})();
